sdonboarding.client/src/components/UITemplate/AddDataTemplate.jsx

```diff
@@ -0,0 +1,304 @@
+
+        //Call the parent function for edit request to server
+        if (valid === true) {
+            await handleAddSubmit();
+            setLocalObject(null);
+        } else {
+            console.log("Form is invalid");
+        }
+        
+    };
+
+     return (
+         <div className='fixed inset-0 bg-gray-400 bg-opacity-70 backdrop-blur-sm  flex justify-center items-center z-50'
+            onClick={handleCancel} //closing the modal if we click outside
+        >
+            <div className="bg-white p-6 rounded shadow-md relative w-[600px] font-semibold"
+                onClick={(e) => e.stopPropagation()} // stopping the component from getting closed when click inside
+            >
+                <h2 className='mb-2'>Create {type} </h2>
+                <hr></hr>
+                <form onSubmit={handleSubmit}>
+                     <div className='mb-4'>
+                         {/*keys of labels are object keys also that we need for creating object*/}
+                        {Object.keys(labels).map((key, index) => {
+                            return (
+                                <div key={key} className="flex flex-col">
+                                    <label htmlFor={key} className='mt-2'>{labels[key]}</label>
+                                    {formElementType[index] === 'boolean' ? (
+                                        // Render dropdown for boolean fields (like 'active')
+                                        <select
+                                            name={key}
+                                            id={key}
+                                            value={localObject[key].toString()}
+                                            onChange={handleActiveChange}
+                                            className="w-full px-4 py-2 border rounded text-gray-400"
+                                            required
+                                        >
+                                            <option value={true}>Active</option>
+                                            <option value={false}>Inactive</option>
+                                        </select>
+                                    ) : formElementType[index] === 'customer' ? (
+                                            // Render dropdown for array
+                                            <select
+                                                name={key}
+                                                id={key}
+                                                value={localObject[key]}
+                                                onChange={(e) => {
+                                                    const selectedOption = e.target.selectedOptions[0];
+                                                    handleListChange(selectedOption.dataset.id, selectedOption.value, selectedOption.dataset.type);
+                                                }}
```

Comment:
Please don't make this a habit. You can create a function for this and configure this variable inside of that function. This way, you can check and verify first if **e.target.selectedOptions** contains value and has items. If you are handling arrays, don't use directly the indexing like **[0] or [any value]** because this is too risky specially when displaying data from the API's response. You should verify and check first if this actually contains data